File: Vanilla/sources/zomboid/zombie/debug/BooleanDebugOption.java
package zombie.debug;

import zombie.config.BooleanConfigOption;
import zombie.core.Core;
import zombie.debug.options.IDebugOption;
import zombie.debug.options.IDebugOptionGroup;

public class BooleanDebugOption extends BooleanConfigOption implements IDebugOption {
   private IDebugOptionGroup m_parent;
   private final boolean m_debugOnly;

   public BooleanDebugOption(String var1, boolean var2, boolean var3) {
      super(var1, var3);
      this.m_debugOnly = var2;
   }

   public boolean getValue() {
      return !Core.bDebug && this.isDebugOnly() ? super.getDefaultValue() : super.getValue();
   }

   public boolean isDebugOnly() {
      return this.m_debugOnly;
   }

   public IDebugOptionGroup getParent() {
      return this.m_parent;
   }

   public void setParent(IDebugOptionGroup var1) {
      this.m_parent = var1;
   }
}

File: Vanilla/sources/zdoc/media/lua/shared/Library/BooleanDebugOption.lua
---@class BooleanDebugOption : zombie.debug.BooleanDebugOption
---@field private m_parent IDebugOptionGroup
---@field private m_debugOnly boolean
BooleanDebugOption = {}

---@public
---@param arg0 IDebugOptionGroup
---@return void
function BooleanDebugOption:setParent(arg0) end

---@public
---@return IDebugOptionGroup
function BooleanDebugOption:getParent() end

---@public
---@return boolean
function BooleanDebugOption:isDebugOnly() end

---@public
---@return boolean
function BooleanDebugOption:getValue() end
